var/spack/repos/builtin/packages/openslide/package.py

```diff
@@ -6,22 +6,36 @@
 from spack.package import *
 
 
-class Openslide(AutotoolsPackage):
+class Openslide(AutotoolsPackage, MesonPackage):
     """OpenSlide reads whole slide image files."""
 
     homepage = "https://openslide.org/"
     url = "https://github.com/openslide/openslide/releases/download/v3.4.1/openslide-3.4.1.tar.xz"
 
     license("LGPL-2.1-only")
 
+    version("4.0.0", sha256="cc227c44316abb65fb28f1c967706eb7254f91dbfab31e9ae6a48db6cf4ae562")
     version("3.4.1", sha256="9938034dba7f48fadc90a2cdf8cfe94c5613b04098d1348a5ff19da95b990564")
 
+    build_system(
+        conditional("meson", when="@4:"), conditional("autotools", when="@3.4.1"), default="meson"
+    )
+
+    with when("build_system=meson"):
+        depends_on("meson@0.53:", type="build")
+
     depends_on("pkgconfig", type="build")
```

Comment:
I'm wondering about the best organization of these dependencies, for both readability and future maintainability. Would you be able to separate them in the following blocks?
```py
    depends_on("pkgconfig", type="build")

    with when("build_system=meson"):
        depends_on("meson@0.53:", type="build")

    with when("build_system=cmake"):
        depends_on("cmake", type="build")

    depends_on("gdk-pixbuf")
    depends_on("jpeg")
    depends_on("libpng")
    depends_on("libtiff@4.0:")
    depends_on("libxml2")

    with when("@4:"):
        depends_on("glib@2.56:")
        depends_on("sqlite@3.14:")

    with when("@3.4.1"):
        depends_on("glib@2.16:")
        depends_on("sqlite@3.6:")
```
(Note: this is **not** the complete list from your version!)

That puts general build dependencies first, then the ones by build systems. Below that the link/run dependencies, again first general and then by version (not build system).

What do you think of this?

Reply:
Hello @wdconinc, no problem at all, thank you for the review and suggestions - I have updated the package recipe with the recommended improvements to dependency organization